--- ruby-ldap-0.9.7/ldap.c.orig	2009-04-19 23:47:23.000000000 +0200
+++ ruby-ldap-0.9.7/ldap.c	2009-04-19 23:48:30.000000000 +0200
@@ -422,8 +420,8 @@
 #ifdef LDAP_OPT_X_TLS
   rb_ldap_define_opt (LDAP_OPT_X_TLS);
 #endif
-#ifdef LDAP_OPT_X_TLS_PROTOCOL
-  rb_ldap_define_opt (LDAP_OPT_X_TLS_PROTOCOL);
+#ifdef LDAP_OPT_X_TLS_PROTOCOL_MIN
+  rb_ldap_define_opt (LDAP_OPT_X_TLS_PROTOCOL_MIN);
 #endif
 #ifdef LDAP_OPT_X_TLS_CIPHER_SUITE
   rb_ldap_define_opt (LDAP_OPT_X_TLS_CIPHER_SUITE);
--- ruby-ldap-0.9.7/conn.c.orig	2009-04-20 00:10:02.000000000 +0200
+++ ruby-ldap-0.9.7/conn.c	2009-04-20 00:10:19.000000000 +0200
@@ -497,8 +497,8 @@
 #ifdef LDAP_OPT_X_TLS_KEYFILE
     case LDAP_OPT_X_TLS_KEYFILE:
 #endif
-#ifdef LDAP_OPT_X_TLS_PROTOCOL
-    case LDAP_OPT_X_TLS_PROTOCOL:
+#ifdef LDAP_OPT_X_TLS_PROTOCOL_MIN
+    case LDAP_OPT_X_TLS_PROTOCOL_MIN:
 #endif
 #ifdef LDAP_OPT_X_TLS_CIPHER_SUITE
     case LDAP_OPT_X_TLS_CIPHER_SUITE:
@@ -627,8 +627,8 @@
 #ifdef LDAP_OPT_X_TLS_KEYFILE
 	case LDAP_OPT_X_TLS_KEYFILE:
 #endif
-#ifdef LDAP_OPT_X_TLS_PROTOCOL
-	case LDAP_OPT_X_TLS_PROTOCOL:
+#ifdef LDAP_OPT_X_TLS_PROTOCOL_MIN
+	case LDAP_OPT_X_TLS_PROTOCOL_MIN:
 #endif
 #ifdef LDAP_OPT_X_TLS_CIPHER_SUITE
 	case LDAP_OPT_X_TLS_CIPHER_SUITE:
